### PR TITLE
Use one param list for first two arguments of ActorContext.ask

### PR DESCRIPTION
**Pull request by patriknw**
I think it would be better to place the first two parameter, which are both about sending the request, into the first parameter list instead of having separate lists for those.

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/ActorContext.scala [hunk 1]
```diff
@@ -286,7 +286,7 @@ trait ActorContext[T] extends TypedActorContext[T] {
    * @tparam Req The request protocol, what the other actor accepts
    * @tparam Res The response protocol, what the other actor sends back
    */
-  def ask[Req, Res](target: RecipientRef[Req])(createRequest: ActorRef[Res] => Req)(
+  def ask[Req, Res](target: RecipientRef[Req], createRequest: ActorRef[Res] => Req)(
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/ActorContext.scala]: Doesn't that mean the Scala compiler can now no longer infer `Req` for the second parameter from the first?

Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/ActorContext.scala]: isn't that covered by some of the test code that I changed?

Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/ActorContext.scala]: Maybe it is never actually used in a way where `createRequest` needs to infer `Req`.

Comment (jrudolph) [reply akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/ActorContext.scala]: Is there a use case for type inference of `Req` given that `RecipientRef` is contravariant? Usually, you already state which message to send. The type that needs to be inferred in most cases is `Res` and that should still work.

File: akka-cluster-typed/src/main/scala/akka/cluster/ddata/typed/javadsl/ReplicatorMessageAdapter.scala [hunk 2]
```diff
@@ -97,11 +98,11 @@ class ReplicatorMessageAdapter[A, B <: ReplicatedData](
   def askUpdate(
       createRequest: JFunction[ActorRef[Replicator.UpdateResponse[B]], Replicator.Update[B]],
       responseAdapter: JFunction[Replicator.UpdateResponse[B], A]): Unit = {
-    context.asScala.ask[Replicator.Update[B], Replicator.UpdateResponse[B]](replicator)(askReplyTo =>
-      createRequest(askReplyTo)) {
-      case Success(value) => responseAdapter(value)
-      case Failure(ex)    => throw ex // unexpected ask timeout
-    }
+    context.asScala
+      .ask[Replicator.Update[B], Replicator.UpdateResponse[B]](replicator, askReplyTo => createRequest(askReplyTo)) {
```
Comment (jrudolph) [inline akka-cluster-typed/src/main/scala/akka/cluster/ddata/typed/javadsl/ReplicatorMessageAdapter.scala]: The explicit type arguments don't seem necessary here (but maybe they are here for documentation purposes). Same at the other instances below.

**Review by jrudolph — APPROVED** [review]
LGTM, from what I've tried there seems to be no functional change with type inference or similar here.

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**